docs/design/features/dotnet-pgo.md

```diff
@@ -0,0 +1,88 @@
+
+```
+usage: dotnet-pgo <command> [<args>]
+
+    create-mibc    Transform a trace file into a Mibc profile data file.
+    merge          Merge multiple Mibc profile data files into one file.
+```
+
+### `create-mibc` command
+Transform a trace file into a Mibc profile data file.
+```
+usage: dotnet-pgo create-mibc --trace-file <arg> --output-file-name <arg> [--pid <arg>] [--process-name [arg]] [--clr-instance-id <arg>] [--reference <arg>...] [--exclude-events-before <arg>]
+                  [--exclude-events-after <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --trace-file <arg>               Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
+    --pid <arg>                      The pid within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified
+    --process-name [arg]             The process name within the trace of the process to examine. If this is a multi-process trace, at least one of --pid or --process-name must be specified.
+    --clr-instance-id <arg>          If the process contains multiple .NET runtimes, the instance ID must be specified.
+    --reference <arg>...             If a reference is not located on disk at the same location as used in the process, it may be specified with a --reference parameter
+    --exclude-events-before <arg>    Exclude data from events before specified time. Time is specified as milliseconds from the start of the trace.
+    --exclude-events-after <arg>     Exclude data from events after specified time. Time is specified as milliseconds from the start of the trace.
+    --verbosity <arg>                Adjust verbosity level.
+    --compressed                     Generate compressed mibc
+    -h, --help                       Display this usage message.
+```
+
+`dotnet-pgo create-mibc` is used to create mibc file from a trace. It supports nettrace, etl, netperf, and netperf.zip trace file formats. The `--trace-file` and `--output-file-name` switches must be specified.
+
+If using trace file format that supports multiple process tracing, specify some combination of the `--pid`, `--process-name` and `--clr-instance-id` switches as needed. When run from the command line in situations where these arguments are needed, the valid args will be printed to the console. Usage of the nettrace file format is recommended for users attempting to run this tool as part of a CI system.
+
+The `dotnet-pgo` tool is designed to run on a machine which has the same file system layout as the trace. If this is not the case, `dotnet-pgo` may not be able to find assemblies loaded by the application. In those cases, the `--reference` switch may be used to inform `dotnet-pgo` where to find assemblies that are part of the application.
+
```

Comment:
Is the argument for `--reference` supposed to be a full path to an assembly, or a directory where the tool will search for missing assemblies, or ...?

Reply:
This argument is a file or path with wild cards. I'll add a note.